radicle-node/src/service.rs

```diff
@@ -891,14 +915,46 @@ where
                         match message.is_fresh(&self.storage) {
                             Ok(is_fresh) => {
                                 if is_fresh {
-                                    // TODO: Only fetch if the refs announced are for peers we're tracking.
-                                    self.fetch(message.rid, announcer);
+                                    match repo_policy.scope {
+                                        tracking::Scope::Trusted => {
+                                            // Check every node and only fetch if we're tracking the node.
+                                            let doc = self.get(message.rid)?;
+
+                                            let trusted: Vec<_> = message.refs.iter().filter_map(|(pub_key, _refs)| {
+                                                let is_delegate = match &doc {
+                                                    Some(d) => d.is_delegate(pub_key),
+                                                    None => false,
+                                                };
+                                                let is_node_tracked = self
+                                                    .tracking.is_node_tracked(pub_key)
+                                                    .expect("handle_announcement: error accessing node tracking configuration");
+                                                if is_node_tracked || is_delegate {
+                                                    Some(*pub_key)
+                                                } else {
+                                                    debug!("handle_announcement: skip fetch as we're not tracking {pub_key}");
```

Comment:
For `expect` error messages, we use the standard `Type::method:` prefix that rust stdlib typically uses, but for logs we include the target only. I've provided examples in other comments.

Reply:
Got it. Thanks!